src/Featurizers/Base/RollingWindowTransformerBase.h

```diff
@@ -0,0 +1,187 @@
+///  \class         RollingWindowTransformerBase
+///  \brief         RollingWindow class that is used for all analytical window computations.
+///                 This class does not have an archive constructor or a save method as those need
+///                 to be implemented by the derived classes.
+///
+template <
+    typename InputT,
+    typename OutputT,
+    size_t MaxNumTrainingItemsV
+>
+class RollingWindowTransformerBase:
+    public Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType = Components::InferenceOnlyTransformerImpl<InputT, std::vector<OutputT>>;
+    using CalculatorFunctionInputType = typename Components::CircularBuffer<InputT>::iterator;
+    using CalculatorFunction = std::function<OutputT(CalculatorFunctionInputType, CalculatorFunctionInputType)>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    RollingWindowTransformerBase(std::uint32_t maxWindowSize, CalculatorFunction calculator, std::uint32_t horizon, std::uint32_t minWindowSize) ;
+    ~RollingWindowTransformerBase(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(RollingWindowTransformerBase);
+
+    bool operator==(RollingWindowTransformerBase const &other) const;
+    bool operator!=(RollingWindowTransformerBase const &other) const;
+
+protected:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Protected Members
+    // |
+    // ----------------------------------------------------------------------
+    // These are protected so that the derived classes can use them for saving of state.
+    const std::uint32_t                             _maxWindowSize;
+    const std::uint32_t                             _minWindowSize;
+    const std::uint32_t                             _horizon;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Members
+    // |
+    // ----------------------------------------------------------------------
+    CalculatorFunction                              _calculator;
+    Components::CircularBuffer<InputT>              _buffer;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {

```

Comment:
>execute_impl [](start = 9, length = 12)

Does this object need `flush_impl` to reset state?

Reply:
Yep. That is on my todo list, but I had forgotten about it this morning. I'll add it in now.

Reply:
`flush_impl` has been added.